proto/zitadel/group/v2/group_service.proto

```diff
@@ -460,4 +508,85 @@ message DeleteGroupResponse {
       example: "\"2025-08-11T15:00:00.051Z\"";
     }
   ];
+}
+
+message AddUsersToGroupRequest {
+  // ID of the group to which the users should be added.
+  string id = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      example: "\"69629012906488334\"";
+    },
+    (google.api.field_behavior) = REQUIRED
+  ];
+
+  // UserIds is a list of IDs of the users who should be added to the group.
+  repeated string user_ids = 2 [
+    (validate.rules).repeated = {min_items: 1, items: {string: {min_len: 1, max_len: 200}}},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[\"69629023906488334\",\"69622366012355662\"]";
+    }
+  ];
+}
+
+message AddUsersToGroupResponse {
+  // FailedUserIDs is the list of IDs of users who could not be added to the group.
+  // todo: do we need to add a failure reason?
+  repeated string failed_user_ids = 1;
+
+  // ChangeDate is the timestamp when the users are added.
+  google.protobuf.Timestamp change_date = 2 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"2025-08-11T15:00:00.051Z\"";
+    }
+  ];
+}
+
+message RemoveUsersFromGroupRequest {
+  // ID of the group from which the users should be removed.
+  string id = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629012906488334\"";
+    },
+    (google.api.field_behavior) = REQUIRED
+  ];
+
+  // UserIds is a list of IDs of the users who should be removed the group.
+  repeated string user_ids = 2 [
+    (validate.rules).repeated = {min_items: 1, items: {string: {min_len: 1, max_len: 200}}},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[\"69629023906488334\",\"69622366012355662\"]";
+    }
+  ];
+}
+
+message RemoveUsersFromGroupResponse {
+  // ChangeDate is the timestamp when the users are removed.
+  google.protobuf.Timestamp change_date = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"2025-08-11T15:00:00.051Z\"";
+    }
+  ];
+}
+
+message ListUsersInGroupRequest {
+  // ID of the group from which users should be retrieved.
+  string id = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629012906488334\"";
+    },
+    (google.api.field_behavior) = REQUIRED
+  ];
+
+  // UserIDs is a list IDs of the users to be retrieved in the provided group.
+  repeated zitadel.filter.v2.InIDsFilter user_ids = 2;
+}
+
+message ListUsersInGroupResponse {
+  // Users is the list of users present in the requested group, matching the search criteria
+  repeated zitadel.authorization.v2beta.User users = 1;
```

Comment:
Define a specific user message for the result as necessary.
As well as include the pagination response.